### PR TITLE
Enable json generator and fork to new repo as new project

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: schema-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParser.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import static javax.json.Json.createReader;
+
+import uk.gov.justice.maven.generator.io.files.parser.FileParser;
+
+
+public class JsonSchemaFileParser implements FileParser<JsonObject> {
+
+    @Override
+    public Collection<JsonObject> parse(Path baseDir, Collection<Path> paths) {
```
Comment (khanhassan) [inline schema-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParser.java]: final?

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.List;
+
+import org.junit.Test;
+
+public class JsonFilesScanningTest {
+
+    @Test
+    public void shouldProcessInternalAndExternalJsonFiles() throws Exception {
+
+        List<String> recordedJsonTitles = Files.readAllLines(recordedJsonTitlesFile());
```
Comment (khanhassan) [inline generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java]: final?

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        assertThat(recordedJsonTitles, hasItems("one", "two"));
+    }
+
+    private Path recordedJsonTitlesFile() throws URISyntaxException {
+        URL classUrl = this.getClass().getResource(this.getClass().getSimpleName() + ".class");
```
Comment (khanhassan) [inline generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java]: final?

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        String generatedSourcesFolder = substringBefore(classUrl.toString(), "test-classes") + "generated-sources/";
```
Comment (khanhassan) [inline generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java]: final?

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/RamlFilesScanningTest.java [hunk 5]
```diff
@@ -17,18 +15,18 @@
 
 import org.junit.Test;
 
-public class RamlFilesScanningIT {
+public class RamlFilesScanningTest {
 
     @Test
     public void shouldProcessInternalAndExternalRamlFiles() throws Exception {
 
         List<String> recordedRamlTitles = Files.readAllLines(recordedRamlTitlesFile());
```
Comment (khanhassan) [inline generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/RamlFilesScanningTest.java]: final?

Comment (khanhassan) [inline generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/RamlFilesScanningTest.java]: final?

File: generator-plugin/src/test/java/uk/gov/justice/maven/generator/io/files/parser/generator/generators/NonInstantiableTestGenerator.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import uk.gov.justice.maven.generator.io.files.parser.core.GeneratorConfig;
+
+import org.raml.model.Raml;
+
+public class NonInstantiableTestGenerator implements Generator<Raml> {
+
+    private NonInstantiableTestGenerator() {}
+
+    @Override
+    public void run(Raml source, GeneratorConfig generatorConfig) {
```
Comment (khanhassan) [inline generator-plugin/src/test/java/uk/gov/justice/maven/generator/io/files/parser/generator/generators/NonInstantiableTestGenerator.java]: final?

File: schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    final Path baseDir = get("src/test/resources/event/");
+    final JsonSchemaFileParser parser = new JsonSchemaFileParser();
+
+    @Test
+    public void shouldParsePathsToJson() throws URISyntaxException {
+
+        final List<Path> paths = asList(
+                get("test-schema1.json"),
+                get("test-schema2.json"));
+        Collection<JsonObject> schemas = parser.parse(
```
Comment (khanhassan) [inline schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java]: final?

File: schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                )
+                )
+        );
+    }
+
+    @Test
+    public void shouldThrowFileNotFoundException() {
+        final List<Path> paths = asList(get("no-schema1.json"));
+        try{
+            Collection<JsonObject> result = parser.parse(baseDir, paths);
```
Comment (khanhassan) [inline schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java]: final?

File: raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.maven.generator.io.files.parser;
+
+import java.nio.file.Path;
+
+import org.raml.model.Raml;
```
Comment (skiddykong) [inline raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java]: done

File: raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+import org.raml.parser.loader.ClassPathResourceLoader;
+import org.raml.parser.loader.FileResourceLoader;
+import org.raml.parser.loader.ResourceLoader;
+import org.raml.parser.visitor.RamlDocumentBuilder;
```
Comment (skiddykong) [inline raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java]: done

File: raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+public class FileHelper {
+    static final String CLASSPATH = "CLASSPATH";
+
+    public FileHelper() {
```
Comment (skiddykong) [inline raml-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/FileHelper.java]: removed

File: schema-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParser.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+        return paths.stream()
+                .map(path -> read(baseDir.resolve(path).toAbsolutePath().toString()))
+                .collect(Collectors.toList());
+    }
+
+    private JsonObject read(final String filePath) {
+        try {
+            return createReader(new FileReader(filePath)).readObject();
```
Comment (skiddykong) [inline schema-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParser.java]: done

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java (hunk 3 above)
Comment (skiddykong) [reply generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java]: done

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java (hunk 4 above)
Comment (skiddykong) [reply generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/JsonFilesScanningTest.java]: done

File: generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/RamlFilesScanningTest.java (hunk 5 above)
Comment (skiddykong) [reply generator-plugin-it/src/test/java/uk/gov/justice/maven/generator/io/files/parser/it/RamlFilesScanningTest.java]: done

File: generator-plugin/src/test/java/uk/gov/justice/maven/generator/io/files/parser/generator/generators/NonInstantiableTestGenerator.java (hunk 6 above)
Comment (skiddykong) [reply generator-plugin/src/test/java/uk/gov/justice/maven/generator/io/files/parser/generator/generators/NonInstantiableTestGenerator.java]: done

File: schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java (hunk 7 above)
Comment (skiddykong) [reply schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java]: done

File: schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java (hunk 8 above)
Comment (skiddykong) [reply schema-parser/src/test/java/uk/gov/justice/maven/generator/io/files/parser/JsonSchemaFileParserTest.java]: done

File: generator-plugin/src/main/java/uk/gov/justice/maven/generator/io/files/parser/generator/GenerateGoalConfig.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+    private final Path outputDirectory;
+
+    private final String basePackageName;
+
+    private final Map<String, String> properties;
+
+    private final List<Path> sourcePaths;
+
+    private GenerateGoalConfig(final String generatorName,
```
Comment (skiddykong) [inline generator-plugin/src/main/java/uk/gov/justice/maven/generator/io/files/parser/generator/GenerateGoalConfig.java]: This is a false positive, the object in question is a configuration

**Pull request merged**